### ISSUE TITLE
install.sh exits without an error message when the download cannot connect

### ISSUE THREAD

**Issue by Sunasuna24** [stoplightio/spectral#3040]
**Describe the bug**

When `scripts/install.sh` cannot reach GitHub at all — DNS failure, connection refused, TLS failure, timeout, offline runner — it terminates with curl's exit code and prints nothing. The error message the script already defines (`Error requesting. Download binary from <URL>`) is never reached, so the user is left with a bare non-zero exit and no indication of what went wrong.

This affects the install path documented in `docs/getting-started/2-installation.md`:

```bash
curl -L https://raw.github.com/stoplightio/spectral/master/scripts/install.sh | sh
```

and the Docker image build, which runs the same script (`Dockerfile:10`).

HTTP-level failures are *not* affected — a 404 or 500 is reported correctly today, because curl exits 0 for those without `-f` and the `if`/`else` below is reached normally. Only connection-level failures are swallowed.

**To Reproduce**

Run the script with no network access:

```bash
docker run --rm --network none --user root -w /tmp \
  -v "$PWD/scripts/install.sh:/install.sh:ro" \
  curlimages/curl:latest sh /install.sh 6.16.3
echo "exit: $?"
```

Actual output:

```
Installing on Alpine Linux.
exit: 6
```

**Expected behavior**

The failure should surface the message the script already contains, and exit 1:

```
Error requesting. Download binary from https://github.com/stoplightio/spectral/releases/download/v6.16.3/spectral-alpine-arm64
exit: 1
```

(the binary filename varies with the host architecture — the run above was on arm64)

**Environment**

- `scripts/install.sh`, all supported platforms
- Reproduced against `develop` @ `0d309d76`

**Additional context**

Three things in the same block combine to swallow the error:

1. **`install.sh:53`** — `set -eu` is active (`install.sh:8`), and `STATUS=$(curl ...)` is a simple command whose exit status is curl's. When curl fails at the connection level it exits non-zero, so the shell terminates on this line and the `if`/`else` below never runs. This is the one that produces the silent `exit 6`.

2. **`install.sh:54`** — `if [ $STATUS -ge 200 ] & [ $STATUS -le 308 ]` uses `&` rather than `&&`. `&` is an asynchronous list separator, so the lower-bound test runs in the background and its result is discarded; the condition is effectively `-le 308` alone. It is currently masked by (1), but becomes reachable the moment (1) is addressed — a status below 200 would then be treated as success.

3. **`install.sh:59`** — `rm "$SRC"` in the `else` branch fails when curl never created the output file, and `set -e` terminates the script before the `echo` on the following line. So even once the `else` branch is reachable, the message can still be lost.

All three need to change together for the `else` branch to actually report anything. I have a fix and will open a PR referencing this issue.
